lib/ansible/modules/network/nxos/nxos_file_copy.py

```diff
@@ -175,12 +176,30 @@
     HAS_PEXPECT = False
 
 
+def md5sum_check(module, dst, file_system):
+    command = 'show file {0}{1} md5sum'.format(file_system, dst)
+    remote_filehash = run_commands(module, {'command': command, 'output': 'text'})[0]
+    remote_filehash = to_bytes(remote_filehash, errors='surrogate_or_strict')
+
+    local_file = module.params['local_file']
+    with open(local_file , 'r') as f:
```

Comment:
Can we do 
try:
    with open(local_file , 'r') as f:
        filecontent = f.read()
except (OSError, IOError) as ioexc:
<>
If we fail to read the file for various OS error like memory error due to very large binary file etc. We can catch the exception and fail the module with informative error.